app/controllers/scimitar/active_record_backed_resources_controller.rb

```diff
@@ -153,18 +153,31 @@ def record_to_scim(record)
       # Save a record, dealing with validation exceptions by raising SCIM
       # errors.
       #
-      # +record+:: ActiveRecord subclass to save (via #save!).
+      # +record+:: ActiveRecord subclass to save.
+      #
+      # If you just let this superclass handle things, it'll call the standard
+      # +#save!+ method on the record. If you pass a block, then this block is
+      # invoked and passed the ActiveRecord model instance to be saved. You can
+      # then do things like calling a different method, using a service object of
+      # some kind, perform audit-related operations and so-on.
       #
       # The return value is not used internally, making life easier for
       # overriding subclasses to "do the right thing" / avoid mistakes (instead
       # of e.g. requiring that a to-SCIM representation of 'record' is returned
       # and relying upon this to generate correct response payloads - an early
       # version of the gem did this and it caused a confusing subclass bug).
       #
-      def save!(record)
-        record.save!
-
+      def save!(record, &block)
+        if block_given?
+          yield(record)
+        else
+          record.save!
+        end
       rescue ActiveRecord::RecordInvalid => exception
+        handle_invalid_record(exception.record)
+      end
+
+      def handle_invalid_record(record)
```

Comment:
This refactor is nice, but I'm of course going to ask for docs :joy: - just a brief RDoc comment, same pattern as `save!`  itself.